test/rubygems/test_gem_commands_signin_command.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+require 'rubygems/test_case'
+require 'rubygems/commands/signin_command'
+require 'rubygems/installer'
+
+class TestGemCommandsSigninCommand < Gem::TestCase
+
+  def setup
+    super
+    
+    Gem.configuration.rubygems_api_key = nil
+    Gem.configuration.api_keys.clear
+    
+    @cmd = Gem::Commands::SigninCommand.new
+  end
+
+  def teardown
+    credentials_path = Gem.configuration.credentials_path
+    File.delete(credentials_path)  if File.exist?(credentials_path)
+    super
+  end
+
+  def test_execute_when_not_already_signed_in
```

Comment:
can these tests also test that the API key is being set in the credentials file properly?
can we also test what happens when signing into a host you're already signed into?